Pokemondetail

AlexanderLill opened the pull request:
Merge the branch with the pokemondetail-page (not entirely finished, but ready enough for showing it to others) into develop.

philbu committed:
Merge remote-tracking branch 'origin/PokeWikiIndex' into pokemondetail

philbu committed:
Merge remote-tracking branch 'origin/PokeWikiIndex' into pokemondetail

philbu committed:
First detail page

philbu committed:
First detail page

AlexanderLill committed:
Add poke-details component files

AlexanderLill committed:
Add poke-details component files

AlexanderLill committed:
Rename pokemonDatato pokemon

AlexanderLill committed:
Rename pokemonDatato pokemon

philbu committed:
No feature page changed

philbu committed:
No feature page changed

AlexanderLill committed:
Add databinding component

AlexanderLill committed:
Add databinding component

AlexanderLill committed:
Finish poke-evolution component

AlexanderLill committed:
Finish poke-evolution component

AlexanderLill committed:
Add description field to pokemon and api.service

AlexanderLill committed:
Fix pokemon field types

AlexanderLill committed:
Fix pokemon field types

AlexanderLill committed:
Finish poke-attack component (no CSS)

AlexanderLill committed:
Finish poke-attack component (no CSS)

AlexanderLill committed:
Merge branch 'pokedescription' into pokemondetail

AlexanderLill committed:
Add pokemon description

AlexanderLill committed:
Add pokemon description

AlexanderLill committed:
Merge pokemondetail with develop

philbu committed:
Type changes

AlexanderLill committed:
Fix poke-detail page

philbu committed:
Merging poketypes with right linking

WoH commented on ionic2/app/components/poke-details/poke-evolutions.component.ts:
```diff
@@ -0,0 +1,40 @@
+        .subscribe(results => this.prevPokemon = results, error => this.prevPokemon = null);
+    }
+
+    for (var i=0, len=this.pokemon.nextEvolutions.length; i<len; i++) {
```
for(evolution of ....)

AlexanderLill replied:
done

WoH commented on ionic2/app/models/pokemon.ts:
```diff
@@ -20,17 +27,17 @@ export class Pokemon {
     breedable: boolean;
   };
 
-  previousEvolutions: [{
```
previousEvolution: Evolution[] or Evolution

AlexanderLill replied:
done

WoH commented on ionic2/app/models/pokemon.ts:
```diff
@@ -20,17 +27,17 @@ export class Pokemon {
-  }];
+  }[];
 
-  nextEvolutions: [{
```
See above

AlexanderLill replied:
done

WoH commented on ionic2/app/pages/poke-detail/poke-detail.page.ts:
```diff
@@ -1,18 +1,29 @@
+  constructor(private navCtrl: NavController, private navParams: NavParams, private apiservice: ApiService) {
     this.pokemonId = navParams.get('pokemonId');
-    console.log('pokemonId', this.pokemonId);
+    this.apiservice.getPokemonById(this.pokemonId).subscribe(results => this.pokemon = results, error => this.pokemon = null);
```
?????

AlexanderLill replied:
done, we push the whole pokemon now

WoH commented on ionic2/app/pages/poke-detail/poke-detail.page.ts:
```diff
@@ -1,18 +1,29 @@
-import {forwardRef} from "@angular/core";
```
Please pull the Pokemon off navParams, don't do another API call.

AlexanderLill replied:
How exactly?
Isn't the poke-detail-page called only with the Id?

`this.navCtrl.push(PokeDetailPage, {pokemonId: pokemon.pokemonId});`

Should we instead push the whole pokemon?

AlexanderLill replied:
done, works :)

philbu committed:
No style for object

johartl commented on ionic2/app/components/poke-details/poke-stats.component.ts:
```diff
@@ -1,11 +1,14 @@
 })
 
 export class PokeStatsComponent {
+  @Input("pokemon") pokemon: Pokemon;
```
Please use single quotes for consistency.

johartl commented on ionic2/app/components/poke-details/poke-type.component.ts:
```diff
@@ -1,12 +1,54 @@
   template: require('./poke-type.component.html'),
-  selector: 'poke-type'
+  selector: 'poke-type',
+  styles: [`
```
I know it's now a lot but I think it's more consistent to move the CSS to a different file rather than inlining it.

johartl commented on ionic2/app/components/poke-details/poke-type.component.ts:
```diff
@@ -1,12 +1,54 @@
+    }
+    this.typeArray = []
+    for (let item of this.type){
+      this.typeArray.push({element:item, color:color[item+'']})
```
What's the reason for using `item+''` here?

johartl commented on ionic2/app/components/poke-details/poke-type.component.ts:
```diff
@@ -1,12 +1,54 @@
+  typeArray:Object[]
 
-  constructor() {}
+  constructor() {
```
If we don't need the constructor just remove it :)

johartl commented on ionic2/app/pages/poke-detail/poke-detail.page.ts:
```diff
@@ -5,14 +5,18 @@ import { ApiService } from '../../services/api.service';
 import { Pokemon } from "../../models/pokemon";
```
Again, would you be able to use single quotes here?

johartl commented on ionic2/app/components/poke-details/poke-stats.component.ts:
```diff
@@ -1,11 +1,14 @@
 export class PokeStatsComponent {
+  @Input("pokemon") pokemon: Pokemon;
 
   constructor() {}
```
Can you remove this constructor please?

johartl commented on ionic2/app/components/poke-details/poke-type.component.ts:
```diff
@@ -0,0 +1,54 @@
+    }
+    this.typeArray = []
+    for (let item of this.type){
+      this.typeArray.push({element:item, color:color[item+'']})
```
What's the reason for using `item+''` here?

AlexanderLill replied:
Phil says there is no reason, it will be removed.

johartl commented on ionic2/app/pages/poke-detail/poke-detail.page.ts:
```diff
@@ -1,18 +1,33 @@
-import {forwardRef} from "@angular/core";
-import {Page, NavController, NavParams} from "ionic-angular";
-import {NavbarComponent} from "../../components/navbar/navbar.component";
+import { forwardRef } from "@angular/core";
```
Do you mind using single quotes throughout the file?

AlexanderLill replied:
replaced ALL of them :)

johartl commented on ionic2/app/components/poke-details/poke-type.component.ts:
```diff
@@ -0,0 +1,54 @@
+@Component({
+  template: require('./poke-type.component.html'),
+  selector: 'poke-type',
+  styles: [`
```
I know it's not a lot of CSS but I think it would still be nice to move it to a separate file rather than inlining it.

AlexanderLill replied:
@mfkaptan  will solve this in his next commit for the attacks styling

johartl commented on ionic2/app/components/poke-details/poke-stats.component.ts:
```diff
@@ -0,0 +1,15 @@
+import { Component, Input } from '@angular/core';
+import { Pokemon } from "../../models/pokemon";
```
Can we use single quotes here? Also for the `@Input("pokemon")` statement below.

AlexanderLill replied:
done

johartl commented on ionic2/app/components/poke-details/poke-evolutions.component.ts:
```diff
@@ -0,0 +1,40 @@
+import { Component, Input } from '@angular/core';
+import { Pokemon } from "../../models/pokemon";
```
Can we use single quotes here?

AlexanderLill replied:
catched them all

AlexanderLill committed:
Make poke-evolutions pretty :3

AlexanderLill committed:
Merge branch 'pokemondetail' of github.com:PokemonGoers/Catch-em-all into pokemondetail

philbu committed:
Poke stats improved

philbu committed:
Merge branch 'pokemondetail' of github.com:PokemonGoers/Catch-em-all into pokemondetail

AlexanderLill committed:
Cleaning; navParams with whole Pokemon; Evolution[]

Removes double quotes
Uses whole pokemon in the navParams to prevent another API call to get the pokemon
Introduce Evolution type

AlexanderLill committed:
Merge pokestats into pokemondetail

WoH commented on ionic2/app/components/poke-details/poke-attacks.component.ts:
```diff
@@ -0,0 +1,13 @@
+
+export class PokeAttacksComponent {
+
+  @Input('pokemon') pokemon: Pokemon;
```
If you don't rename you can omit the selector:
@Input() pokemon: Pokemon;
I saw this a few times throughout and thought it might be helpful to point it out.

AlexanderLill replied:
done, thanks :)

WoH commented on ionic2/app/components/poke-details/poke-evolutions.component.ts:
```diff
@@ -0,0 +1,40 @@
+
+  @Input('pokemon') pokemon: Pokemon;
+
+  public prevPokemon: Pokemon;
```
Since the model defines Evolution[] as prev Pokemon it would be consistent to make this Pokemon[]

AlexanderLill replied:
done

WoH commented on ionic2/app/components/poke-details/poke-evolutions.component.ts:
```diff
@@ -0,0 +1,40 @@
+  selector: 'poke-evolutions'
+})
+
+export class PokeEvolutionsComponent {
```
Angular 2 Styleguide would ask you to import and implement the OnInit interface if you use ngOnInit

AlexanderLill replied:
done

WoH commented on ionic2/app/components/poke-details/poke-type.component.ts:
```diff
@@ -0,0 +1,55 @@
+  }
+
+  ngOnInit(){
+    const color={
```
It would be nice if we could move this out of the component

mfkaptan committed:
Style improvement on pokemon detail
- Fix badges and colors
- Center align badges
- Add cards for beautiful view
- Move "gender" to top
- Remove `rounded-edges` style

mfkaptan committed:
Merge branch 'pokedex-detail-improvement' into pokemondetail

AlexanderLill committed:
Implement onSearch function for wikisearch

AlexanderLill committed:
Implement feedback and improve code style

AlexanderLill committed:
Merge branch 'develop' into pokemondetail

AlexanderLill committed:
Merge pokemondetail with develop
